### PR TITLE
Adding composite implicit version

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #76834
* __->__ #76833
* #76832

fix attn_backward

### CODE REVIEW

File: aten/src/ATen/native/Activation.cpp [hunk 1]
```diff
@@ -424,6 +425,14 @@ Tensor hardswish_backward(const Tensor& grad_output, const Tensor& self) {
   return iter.output();
 }
 
+std::tuple<Tensor, Tensor> attn(const Tensor & q, const Tensor & k, const Tensor & v){
+  auto k_t = at::transpose(k, 0, 1);
```
Comment (albanD) [inline aten/src/ATen/native/Activation.cpp:429]: I would expect to see some TORCH_CHECK to make sure the size of the inputs and their number of dimensions is what you expect!

**Review by albanD — APPROVED** [review]
SGTM
For a real function I would expect more error checking but good here.

File: aten/src/ATen/native/Activation.cpp [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #include <ATen/NativeFunctions.h>
 #include <ATen/native/TensorIterator.h>
 #include <ATen/Parallel.h>
+#include <ATen/ops/matmul.h>
```
Comment (soulitzer) [inline aten/src/ATen/native/Activation.cpp:9]: Do you actually need this?

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -17172,6 +17182,16 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
         supports_forward_ad=True,
         supports_fwgrad_bwgrad=True,
         supports_out=False),
+    OpInfo(
+        "attn",
+        ref=_NOTHING,
```
Comment (soulitzer) [inline torch/testing/_internal/common_methods_invocations.py:17187]: Is _NOTHING the default here as well?

Comment (drisspg) [reply torch/testing/_internal/common_methods_invocations.py:17235]: will remove

File: aten/src/ATen/native/Activation.cpp (hunk 2 above)
Comment (drisspg) [reply aten/src/ATen/native/Activation.cpp:9]: tried with out and safe to remove

**Pull request closed without merging**